oracle/pkg/l1Listener/l1Listener.go

```diff
@@ -159,29 +179,54 @@ func (l *L1Listener) watchL1Block(ctx context.Context) error {
 				l.logger.Error("failed to get block number", "error", err)
 				continue
 			}
+			// We need to get the previous block number because the current block has finalized header
+			blockNum = blockNum - 1
 
 			if blockNum <= uint64(currentBlockNo) {
 				continue
 			}
 
 			for b := uint64(currentBlockNo) + 1; b <= blockNum; b++ {
-				header, err := l.l1Client.HeaderByNumber(ctx, big.NewInt(int64(b)))
+				header, err := l.l1Client.HeaderByNumber(ctx, big.NewInt(int64(b+1)))
 				if err != nil {
 					l.logger.Error("failed to get header", "block", b, "error", err)
 					continue
 				}
 
-				winner := string(bytes.ToValidUTF8(header.Extra, []byte("�")))
+				// End of changes needed to be done.
+				var builderPubKey string
+				startTime := time.Now()
+				l.logger.Info("querying relay", "block", b, "hash", header.ParentHash.Hex())
+				for time.Since(startTime) < 2*time.Second {
```

Comment:
why we are trying for 2 seconds and not for any other time?

Reply:
It seemed to work well, there's no strong reason.

Reply:
It's just to attempt around 4 retries.